extension/source/QuillPage/components/OnboardingActionPanel.tsx

```diff
@@ -13,17 +13,30 @@ const OnboardingActionPanel: React.FunctionComponent<{ pageIndex: number }> = ({
     <WorkflowNumbers
       current={pageIndex + 1}
       max={3}
-      onSelect={(pageNumber) => {
-        // Note: This is for demo purposes only. We're not going to be using
-        // search parameters like this.
-        window.location.href = `?p=${pageNumber}`;
+      onSelect={(workflowNumber) => {
+        window.location.href = `?p=${workflowNumber}`;
       }}
     />
     {
       [
-        <PasswordCreationPanel key={1} />,
-        <SetNicknamePanel key={2} />,
-        <ViewSecretPhrasePanel key={3} />,
+        <PasswordCreationPanel
+          key={1}
+          onComplete={() => {
+            window.location.href = `?p=2`;
+          }}
```

Comment:
```suggestion
          next={2}
          }}
```
How about instead of passing a whole function as prop, just pass in the next page number and handle onComplete inside the target component? Prop passing should be kept as light as possible and individual components should remain as independent as possible.

Reply:
I think this is the same point as https://github.com/jzaki/bls-wallet/pull/93#discussion_r784374486. If so, let's followup there.